app/javascript/components/Admin/Products/Purchases/Content.tsx

```diff
@@ -0,0 +1,53 @@
+import React from "react";
+
+import Loading from "$app/components/Admin/Loading";
+
+import AdminProductPurchase, { ProductPurchase } from "./Purchase";
+
+type AdminProductPurchasesContentProps = {
+  purchases: ProductPurchase[];
+  isLoading: boolean;
+  hasMore: boolean;
+  onLoadMore: () => void;
+};
+
+const AdminProductPurchasesContent = ({
+  purchases,
+  isLoading,
+  hasMore,
+  onLoadMore,
+}: AdminProductPurchasesContentProps) => {
+  const handleClick = (e: React.MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+    if (hasMore) {
+      onLoadMore();
+    }
+  };
+
+  if (purchases.length === 0 && !isLoading)
+    return (
+      <div className="info" role="status">
+        No purchases have been made.
+      </div>
+    );
+
+  return (
+    <div className="paragraphs">
+      <div className="stack">
+        {purchases.map((purchase) => (
+          <AdminProductPurchase key={purchase.id} purchase={purchase} />
+        ))}
+      </div>
+
+      {isLoading ? <Loading /> : null}
+
+      {hasMore ? (
+        <button className="button small" onClick={handleClick} disabled={isLoading}>
+          {isLoading ? "Loading..." : "Load more"}
+        </button>
+      ) : null}
```

Comment:
We can use infinite scroll via `useIsIntersecting` here if we want to spare the user from having to click

Reply:
I think this pattern is better UX in the expandable sections (at least for things like purchases where there may be a vast number) so that in search results you can still scroll down to the next result